lib/telemetry.go

```diff
@@ -221,6 +220,10 @@ func (c *TelemetryConfig) MergeDefaults(defaults *TelemetryConfig) {
 		// implementing this for the types we actually have for now. Test failure
 		// should catch the case where we add new types later.
 		switch f.Kind() {
+		case reflect.Struct:
+			if f.Type() == reflect.TypeOf(prometheus.PrometheusOpts{}) {
+				continue
+			}
```

Comment:
I couldn't find any callers of `TelemetryConfig.MergeDefaults` in OSS or Ent. I suspect this `MergeDefaults` function is leftover from a while ago. The only place I see it called is a test, which is I guess what failed and prompted this change. It looks like the only non-test call to it was removed in 65be58703cfb4a19d8fb7fc81228dabc8bd407de

Not urgent right now, but in a follow up we could delete this method I think.

Reply:
Adding a note comment